framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -593,195 +484,306 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
                     return;
                 }
 
-                standardizeVariable(declarationTree, type, elt);
+                atVariableDeclaration(declarationTree, type, elt);
                 return;
 
             default:
-                // It's not a variable (it might be METHOD, CONSTRUCTOR, CLASS, or INTERFACE, for
+                // It's not a local variable (it might be METHOD, CONSTRUCTOR, CLASS, or INTERFACE,
+                // for
                 // example), so there is nothing to do.
                 break;
         }
     }
 
     /**
-     * Parse and standardize the expressions in dependent types in {@code type} as if they were
-     * written at {@code localVarPath}. For example, {@code @KeyFor("field") String} is changed to
-     * {@code @KeyFor("this.field")}
+     * Viewpoint-adapt a type qualifier obtained from a contract using {@code stringToJavaExpr}.
      *
-     * @param localVarPath the expression is parsed as if it were written at this location
-     * @param type the type to parse; is side-effected by this method
+     * <p>For example, if the contract is {@code @EnsuresKeyFor(value = "this.field", map =
```

Comment:
Equally relevant is that in this case the argument corresponding to formal parameter `annoFromContract` is `@KeyFor("this.map")`, so I would mention that too.